kaldb/src/main/java/com/slack/kaldb/server/RecoveryTaskCreator.java

```diff
@@ -147,11 +148,25 @@ public List<SnapshotMetadata> deleteStaleLiveSnapshots(List<SnapshotMetadata> sn
    */
   public long determineStartingOffset(long currentHeadOffsetForPartition) {
     // Filter stale snapshots for partition.
+    if (partitionId == null) {
+      LOG.warn("PartitionId can't be null.");
+    }
+
     List<SnapshotMetadata> snapshots = snapshotMetadataStore.listSync();
     List<SnapshotMetadata> snapshotsForPartition =
         snapshots
             .stream()
-            .filter(snapshotMetadata -> snapshotMetadata.partitionId.equals(partitionId))
+            .filter(
+                snapshotMetadata -> {
+                  if (snapshotMetadata == null || snapshotMetadata.partitionId == null) {
+                    LOG.warn(
+                        "snapshot metadata or partition id can't be null: "
+                            + Strings.join(snapshots, ','));
```

Comment:
This should be `String.join` - then we also won't need the import statement.

Reply:
Done.